src/Domain/Creators/TimeDependence/UniformRotationAboutZAxis.cpp

```diff
@@ -0,0 +1,148 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Creators/TimeDependence/UniformRotationAboutZAxis.hpp"
+
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <ostream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/IndexType.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/Identity.hpp"
+#include "Domain/CoordinateMaps/MapInstantiationMacros.hpp"
+#include "Domain/CoordinateMaps/TimeDependent/ProductMaps.hpp"
+#include "Domain/CoordinateMaps/TimeDependent/ProductMaps.tpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace domain {
+namespace creators {
+namespace time_dependence {
+
+template <size_t MeshDim>
+UniformRotationAboutZAxis<MeshDim>::UniformRotationAboutZAxis(
+    const double initial_time, const double angular_velocity,
+    std::string function_of_time_name) noexcept
+    : initial_time_(initial_time),
+      angular_velocity_(angular_velocity),
+      function_of_time_name_(std::move(function_of_time_name)) {}
+
+template <size_t MeshDim>
+std::unique_ptr<TimeDependence<MeshDim>>
+UniformRotationAboutZAxis<MeshDim>::get_clone() const noexcept {
+  return std::make_unique<UniformRotationAboutZAxis>(
+      initial_time_, angular_velocity_, function_of_time_name_);
+}
+
+template <size_t MeshDim>
+std::vector<std::unique_ptr<
+    domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, MeshDim>>>
+UniformRotationAboutZAxis<MeshDim>::block_maps(
+    const size_t number_of_blocks) const noexcept {
+  ASSERT(number_of_blocks > 0, "Must have at least one block to create.");
+  std::vector<std::unique_ptr<
+      domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, MeshDim>>>
+      result{number_of_blocks};
+  result[0] = std::make_unique<MapForComposition>(map_for_composition());
+  for (size_t i = 1; i < number_of_blocks; ++i) {
+    result[i] = result[0]->get_clone();
+  }
+  return result;
+}
+
+template <size_t MeshDim>
+std::unordered_map<std::string,
+                   std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+UniformRotationAboutZAxis<MeshDim>::functions_of_time() const noexcept {
```

Comment:
I don't quite understand the purpose of this function, but that's probably because I'm not familiar enough with the rest of the moving mesh code. Still, adding some documentation to the declaration could help

Reply:
This is inherited from the base class, so here's not the place to document it

Reply:
I think the function is clearly documented in the base class, `TimeDependence.hpp:89`. It just returns an unordered_map containing the functions of time used by the time dependence to make a domain time dependent.